src/Symfony/Component/Serializer/Encoder/ChainDecoder.php

```diff
@@ -32,6 +32,8 @@ public function __construct(array $decoders = array())
 
     /**
      * {@inheritdoc}
+     * 
+     * @throws RuntimeException If no decoder is found.
```

Comment:
Simple, instead of doing `catch (\Exception $e)` I can use `catch (SerializerException $e)` and have different things in each catch block. How does that break encapsulation?

I mean, that changes simply says that the serializer & co. can throw this type of exception, which is already the current behaviour (but undocumented).